comms/core/src/protocol/rpc/server/mod.rs

```diff
@@ -749,12 +748,17 @@ where
         let mut stream = body
             .into_message()
             .map(|result| into_response(request_id, result))
-            .flat_map(move |message| {
+            .map(move |message| {
                 #[cfg(feature = "metrics")]
                 if !message.status.is_ok() {
                     metrics::status_error_counter(&node_id, &protocol, message.status).inc();
                 }
-                stream::iter(ChunkedResponseIter::new(message))
+                proto::rpc::RpcResponse {
+                    request_id,
+                    status: message.status.as_u32(),
+                    flags: message.flags.bits().into(),
+                    payload: message.payload.to_vec(),
+                }
```

Comment:
I didn't notice the `message.to_proto()` function before. That would be better to use than this

```suggestion
               message.to_proto()
```